Delete AT_WARN and replace all AT_WARN with TORCH_WARN

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34623 Delete AT_WARN and replace all AT_WARN with TORCH_WARN**

The bandaid of "AT_WARN" keeps introducing new warnings. Let's get rid
of it entirely.

Close #34502

Differential Revision: [D20420112](https://our.internmc.facebook.com/intern/diff/D20420112)

xuhdev committed:
Delete AT_WARN and replace all AT_WARN with TORCH_WARN

The bandaid of "AT_WARN" keeps introducing new warnings. Let's get rid
of it entirely.

Close #34502

[ghstack-poisoned]

albanD commented on aten/src/ATen/native/cudnn/RNN.cpp:
```diff
@@ -1219,7 +1219,7 @@ Tensor try_get_weight_buf(
 }
 
-const char * WEIGHT_FORMAT_WARN = "RNN module weights are not part of single contiguous "
+const char * WEIGHT_FORMTORCH_WARN = "RNN module weights are not part of single contiguous "
```
I think this is a spew from the replace ;)

xuhdev replied:
Good catch!

xuhdev committed:
Update on "Delete AT_WARN and replace all AT_WARN with TORCH_WARN"


The bandaid of "AT_WARN" keeps introducing new warnings. Let's get rid
of it entirely.

Close #34502

[ghstack-poisoned]

albanD reviewed (approved):
Thanks !
